src/Camera.cc

```diff
@@ -25,70 +18,38 @@ Camera::Camera(CameraType camtype) :
 void Camera::setup(std::vector<double> params)
 {
 	if (params.size() == 7) {
-		type = CameraType::GIMBAL;
-		object_trans << params[0], params[1], params[2];
-		object_rot << params[3], params[4], params[5];
+		setVpt(params[0], params[1], params[2]);
+		setVpr(params[3], params[4], params[5]);
 		viewer_distance = params[6];
 	} else if (params.size() == 6) {
-		type = CameraType::VECTOR;
-		eye << params[0], params[1], params[2];
-		center << params[3], params[4], params[5];
+		Eigen::Vector3d eye(params[0], params[1], params[2]);
+		Eigen::Vector3d center(params[3], params[4], params[5]);
+		object_trans = -center;
+		auto dir = center - eye;
```

Comment:
Technically, both eye, center and dir are `const`.

Reply:
OK